ci/treefmt: add biome

wolfgangwalther opened the pull request:
Adds biome as a formatter, for now only for everything except `pkgs/*`. There are too many files to go through in there, right now.

Avoids formatting for the CI scripts from getting out of sync all the time, see https://github.com/NixOS/nixpkgs/pull/434541#discussion_r2287982452.

## Things done

- [x] Fits [CONTRIBUTING.md], [pkgs/README.md], [maintainers/README.md] and other READMEs.

[NixOS tests]: https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests
[Package tests]: https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests
[nixpkgs-review usage]: https://github.com/Mic92/nixpkgs-review#usage

[CONTRIBUTING.md]: https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md
[lib/tests]: https://github.com/NixOS/nixpkgs/blob/master/lib/tests
[maintainers/README.md]: https://github.com/NixOS/nixpkgs/blob/master/maintainers/README.md
[nixos/tests]: https://github.com/NixOS/nixpkgs/blob/master/nixos/tests
[pkgs/README.md]: https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md
[pkgs/test]: https://github.com/NixOS/nixpkgs/blob/master/pkgs/test

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

wolfgangwalther committed:
ci/treefmt: add biome for .js files

This excludes doc and pkgs folders, because there are way too many files
to fix at the moment.

wolfgangwalther committed:
ci: apply safe formatting with biome

wolfgangwalther committed:
ci: apply unsafe fixes with biome

wolfgangwalther committed:
doc: apply safe formatting with biome

wolfgangwalther committed:
doc: fix biome linting errors

wolfgangwalther committed:
ci/treefmt: enable biome for doc/

MattSturgeon commented on ci/default.nix:
```diff
@@ -42,6 +42,22 @@ let
+        };
+        settings.formatter.biome.excludes = [
+          "*.min.js"
+          "pkgs/*"
```
Interesting that this apparently doesn't need `pkgs/**`?

Makes me wonder if this is _actually_ matching any part of the path (effectively `**/pkgs/*/**`, with the leading `**/` and trailing `/**` added implicitly by treefmt)?

If so, I'd expect `pkgs` to work without any `/*` needed.

cc @jfly @brianmcgee

MattSturgeon commented on doc/style.css:
```diff
@@ -1,267 +1,267 @@
 html {
-    line-height: 1.15;
-    -webkit-text-size-adjust: 100%;
+	line-height: 1.15;
+	-webkit-text-size-adjust: 100%;
```
spaces -> tabs could be controversial and seems an unnecessary diff (IMO).

Could we configure biome to either ignore css for now, or use spaces?

What (if anything) do we have configured in our editorconfig for css indentation?

MattSturgeon reviewed (commented):
JS looks good, not convinced about the CSS indentation though.

wolfgangwalther committed:
.editorconfig: indent css with spaces

We already do in doc/style.css, but we should tell our editors and
formatters, too.

The comment.. doesn't really make sense, it's purely repetitive with the
code itself. Thus, removed.
